algorithmic_efficiency/workloads/ogb/ogb_jax/workload.py

```diff
@@ -0,0 +1,211 @@
+    del update_batch_norm  # No BN in the GNN model.
+    assert model_state is None
+    train = mode == spec.ForwardPassMode.TRAIN
+    logits = self._model.apply({'params': params},
+                               augmented_and_preprocessed_input_batch,
+                               rngs={'dropout': rng},
+                               train=train)
+    return logits, None
+
+  def _binary_cross_entropy_with_mask(self,
+                                      labels: jnp.ndarray,
+                                      logits: jnp.ndarray,
+                                      mask: jnp.ndarray) -> jnp.ndarray:
+    """Binary cross entropy loss for logits, with masked elements."""
+    assert logits.shape == labels.shape == mask.shape
+    assert len(logits.shape) == 2
+
+    # To prevent propagation of NaNs during grad().
+    # We mask over the loss for invalid targets later.
+    labels = jnp.where(mask, labels, -1)
+
+    # Numerically stable implementation of BCE loss.
+    # This mimics TensorFlow's tf.nn.sigmoid_cross_entropy_with_logits().
+    positive_logits = (logits >= 0)
+    relu_logits = jnp.where(positive_logits, logits, 0)
+    abs_logits = jnp.where(positive_logits, logits, -logits)
+    return relu_logits - (logits * labels) + (jnp.log(1 + jnp.exp(-abs_logits)))
+
+  # Does NOT apply regularization, which is left to the submitter to do in
+  # `update_params`.
+  def loss_fn(
+      self,
+      label_batch: spec.Tensor,
+      logits_batch: spec.Tensor,
+      mask_batch: Optional[spec.Tensor]) -> spec.Tensor:  # differentiable
+    loss = self._binary_cross_entropy_with_mask(
+        labels=label_batch, logits=logits_batch, mask=mask_batch)
+    mean_loss = jnp.sum(jnp.where(mask_batch, loss, 0)) / jnp.sum(mask_batch)
+    return mean_loss
+
+  def _eval_metric(self, labels, logits, masks):
+    loss = self.loss_fn(labels, logits, masks)
+    return metrics.EvalMetrics.single_from_model_output(
+        loss=loss, logits=logits, labels=labels, mask=masks)
+
+  @functools.partial(
+      jax.pmap,
+      axis_name='batch',
+      in_axes=(None, 0, 0, 0, 0, 0, None),
+      static_broadcasted_argnums=(0,))
+  def _eval_batch(self, params, graphs, labels, masks, model_state, rng):
+    logits, _ = self.model_fn(
+        params,
+        graphs,
+        model_state,
+        spec.ForwardPassMode.EVAL,
+        rng,
+        update_batch_norm=False)
+    return self._eval_metric(labels, logits, masks)
+
+  def eval_model(self,
+                 params: spec.ParameterContainer,
+                 model_state: spec.ModelAuxiliaryState,
+                 rng: spec.RandomState,
+                 data_dir: str):
+    """Run a full evaluation of the model."""
+    data_rng, model_rng = prng.split(rng, 2)
+    eval_per_core_batch_size = 1024
+    if self._eval_iterator is None:
+      self._eval_iterator = self._build_iterator(
+          data_rng, 'validation', data_dir, batch_size=eval_per_core_batch_size)
```

Comment:
Do we want to use the `validation` or `test` split here? For the validation split the mAP is >25 in the Flax example (25.2 without and 25.7 with early stopping).

Reply:
We should probably run on both, and append `split/` to make the keys `split/metric_name`. I'll do that in a follow-up CL for all workloads.